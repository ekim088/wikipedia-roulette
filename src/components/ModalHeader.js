// @flow
import * as React from 'react';
import { useContext } from 'react';
import useEventCallback from '@restart/hooks/useEventCallback';
import ModalContext from './ModalContext';

type Props = {
	children?: React.Node | string,
	closeButton?: boolean,
	onHide?: (?Event) => void
};

const ModalHeader = ({ children, closeButton, onHide }: Props) => {
	/**
	 * Propogate `onHide` call to method located on parent context in addition
	 * to local if available. Supports <Modal.Header> renders within <Modal>
	 */
	const context = useContext(ModalContext);
	const handleClick = useEventCallback(() => {
		if (context) context.onHide();
		if (onHide) onHide();
	});

	return (
		<header className="wa-modal-header">
			{children}
			{closeButton && (
				<button type="button" className="wa-modal-close" onClick={handleClick}>
					X
				</button>
			)}
		</header>
	);
};

ModalHeader.defaultProps = {
	children: undefined,
	closeButton: false,
	onHide: undefined
};

export default ModalHeader;
