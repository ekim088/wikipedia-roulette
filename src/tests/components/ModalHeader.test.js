import ModalHeader from '../../components/ModalHeader';

describe('ModalHeader', () => {
	it('should render a modal header', () => {
		const wrapper = shallow(<ModalHeader />);
		expect(wrapper).toMatchSnapshot();
	});

	it('should render a modal header with a close button', () => {
		const wrapper = shallow(<ModalHeader closeButton />);
		expect(wrapper).toMatchSnapshot();
	});

	it('should call onHide on click of the close button', () => {
		const mockClickHandler = jest.fn();
		shallow(<ModalHeader closeButton onHide={mockClickHandler} />)
			.find('button')
			.simulate('click');
		expect(mockClickHandler).toHaveBeenCalled();
	});

	it('should not error on click of the close button without onHide', () => {
		const wrapper = shallow(<ModalHeader closeButton />);
		expect(() => {
			wrapper.find('button').simulate('click');
		}).not.toThrow();
	});
});
