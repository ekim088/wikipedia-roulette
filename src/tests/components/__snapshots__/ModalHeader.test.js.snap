// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ModalHeader should render a modal header 1`] = `
<header
  className="wa-modal-header"
/>
`;

exports[`ModalHeader should render a modal header with a close button 1`] = `
<header
  className="wa-modal-header"
>
  <button
    className="wa-modal-close"
    onClick={[Function]}
    type="button"
  >
    X
  </button>
</header>
`;
